recap/clients/postgresql.py

```diff
@@ -78,3 +79,25 @@ def ls_catalogs(self) -> list[str]:
             """
         )
         return [row[0] for row in cursor.fetchall()]
+
+    def schema(self, catalog: str, schema: str, table: str) -> StructType:
+        cursor = self.connection.cursor()
+        cursor.execute(
+            f"""
+                SELECT
+                    information_schema.columns.*,
+                    pg_attribute.attndims
+                FROM information_schema.columns
+                JOIN pg_attribute on information_schema.columns.column_name = pg_attribute.attname
```

Comment:
I don't think this is enough. I think you need to join on table, schema, and column. It appears `pg_attribute` doesn't have those:

```
               Table "pg_catalog.pg_attribute"
     Column     |   Type    | Collation | Nullable | Default 
----------------+-----------+-----------+----------+---------
 attrelid       | oid       |           | not null | 
 attname        | name      |           | not null | 
 atttypid       | oid       |           | not null | 
 attstattarget  | integer   |           | not null | 
 attlen         | smallint  |           | not null | 
 attnum         | smallint  |           | not null | 
 attndims       | integer   |           | not null | 
 attcacheoff    | integer   |           | not null | 
 atttypmod      | integer   |           | not null | 
 attbyval       | boolean   |           | not null | 
 attalign       | "char"    |           | not null | 
 attstorage     | "char"    |           | not null | 
 attcompression | "char"    |           | not null | 
 attnotnull     | boolean   |           | not null | 
 atthasdef      | boolean   |           | not null | 
 atthasmissing  | boolean   |           | not null | 
 attidentity    | 
"char"    |           | not null | 
 attgenerated   | "char"    |           | not null | 
 attisdropped   | boolean   |           | not null | 
 attislocal     | boolean   |           | not null | 
 attinhcount    | integer   |           | not null | 
 attcollation   | oid       |           | not null | 
 attacl         | aclitem[] |           |          | 
 attoptions     | text[]    | C         |          | 
 attfdwoptions  | text[]    | C         |          | 
 attmissingval  | anyarray  |           |          | 
```

So perhaps we need another join here as well?

Per-ChatGPT:

![Screenshot 2023-12-23 at 11 52 44 AM](https://github.com/recap-build/recap/assets/167569/329c92e0-03c3-4e61-a93f-b5d9f855f496)

Reply:
Ah nice catch, I'll add in the other joins